nifi-nar-bundles/nifi-geohash-bundle/nifi-geohash-processors/src/main/java/org/apache/nifi/processors/geohash/GeohashRecord.java

```diff
@@ -0,0 +1,449 @@
+                                && updateRecord(LONGITUDE_RECORD_PATH, String.valueOf(decodedPoint.getLongitude()), record, paths);
+                    }
+                }
+
+                if (routingStrategy == RoutingStrategy.REQUIRE_ALL_ENRICHED) {
+                    if (!updated) { //If the routing strategy is Require_All_Enriched and there exists a record that is not updated, the entire flowfile should be route to REL_FAILURE
+                        targetRelationship = REL_FAILURE;
+                    }
+                } else {
+                    if (updated) {
+                        //If the routing strategy is Skip_Unenriched and there exists a record that is updated, the entire flowfile should be route to REL_SUCCESS
+                        //If the routing strategy is Split_Enriched_Unenriched and there exists a record that is updated, this record should be route to REL_SUCCESS
+                        targetRelationship = REL_SUCCESS;
+                    }
+                }
+
+                if (routingStrategy != RoutingStrategy.SPLIT || updated) {
+                    writer.write(record);
+                    foundCount++;
+                } else { //if the routing strategy is Split_Enriched_Unenriched and the record is not updated
+                    notFoundWriter.write(record);
+                    notFoundCount++;
+                }
+            }
+
+            final WriteResult writeResult = writer.finishRecordSet();
+            writer.close();
+
+            WriteResult notFoundWriterResult = null;
+
+            if (notFoundWriter != null) {
+                notFoundWriterResult = notFoundWriter.finishRecordSet();
+                notFoundWriter.close();
+            }
+
+            is.close();
+            os.close();
+
+            if (osNotFound != null) {
+                osNotFound.close();
+            }
+
+            output = session.putAllAttributes(output, buildAttributes(foundCount, writer.getMimeType(), writeResult));
+            if (routingStrategy != RoutingStrategy.SPLIT) {
+                session.transfer(output, targetRelationship);
+                session.remove(input);
+            } else {
+                if (notFoundCount > 0) {
+                    notFound = session.putAllAttributes(notFound, buildAttributes(notFoundCount, writer.getMimeType(), notFoundWriterResult));
+                    session.transfer(notFound, REL_FAILURE);
+                } else {
+                    session.remove(notFound);
+                }
+                session.transfer(output, REL_SUCCESS);
+                session.transfer(input, REL_ORIGINAL);
+            }
+
+        } catch (Exception ex) {
+            getLogger().error("Failed to {} due to {}", encode ? "encode" : "decode", ex.getLocalizedMessage(), ex);
+        }
+    }
+
+
+    private Object getEncodedGeohash(RecordPath latitudePath, RecordPath longitudePath, Record record, String format, int level) {
```

Comment:
Recommend changing `format` to `GeohashFormat` instead of `String` to avoid calling `GeohashFormat.valueOf()` for each record evaluation.
```suggestion
    private Object getEncodedGeohash(RecordPath latitudePath, RecordPath longitudePath, Record record, GeohashFormat format, int level) {
```